resources/design-document.md

```diff
@@ -4,188 +4,185 @@
-* Accepts GET requests to /warehouses/id
+## 5.6 Get Warehouse Endpoint
+* Accepts GET requests to /warehouses/:warehouseId
 * Accepts a warehouseId and user info, and returns the relevant warehouseModel
-
-## 6.8 Delete Warehouse Endpoint
-* Accepts DELETE requests to /warehouses/id
-* Accepts a warehouseId and user info, and deletes the matching warehouse, and shipping data
-
-## 6.9 Update Warehouse Inventory Endpoint
-* Accepts PUT requests to /warehouses/id
-* Accepts a warehouseId, itemId and user info
-* Updates warehouse inventory count and returns warehouseModel
-
-## 6.10 Get WareHouse Inventory Report
-* Accepts GET requests to warehouses/id/inventory
-* Accepts a warehouseId and returns a warehouseModel
-* Calculates current inventory status for current warehouse inventory items, and pulls item data
-* Calculates item data and recommends new items to be added to the warehouse, and items to be removed
-* Returns a JSON formatted report to be formatted in the front end
-
-## 6.11 Get WareHouse Shipping Report
- * Accepts GET requests to shipped/warehouseId/{date-range}
- * Accepts a warehouseID and returns a warehouseModel
- * Generates a JSON formatted report of shipping activity within the date-range to be formatted by the front end
-
-## 6.12 Get WareHouse Item Shipping Report
-* Accepts GET requests to shipped/warehouseID/itemId
-* Accepts a warehouseId and itemID and returns a list of shippedModel objects
-
-
-
-# 7. Tables
-
-## 7.1. warehouses
+  * if the warehouseId is not found, will throw 'WarehouseNotFoundException'
+
+## 5.7 Get All Warehouses Endpoint
+* Accepts GET requests to /warehouses
+* Accepts user information and returns a list all warehouses associated with the account
+
+## 5.8 Delete Warehouse Endpoint
+* Accepts DELETE requests to /warehouses/:warehouseId
+* Accepts a warehouseId and user info, and deletes the matching warehouse, inventory, and transactions
+  *  if the warehouseId is not found, will throw 'WarehouseNotFoundException'
+
+## 5.9 Update Warehouse Endpoint
+* Accepts PUT requests to /warehouses/:warehouseId
+* Accepts a warehouseId, allows the user to change the warehouse name, and returns a warehouseModel
+  * if the warehouseId is not found, will throw 'WarehouseNotFoundException'`
+
+## 5.10 Create Transaction Endpoint
+* Accepts a post request to /warehouses/:warehouseId/transactions
+* Accepts a map of InventoryItems to integers
+* Creates a transaction table entry for each item and adjusts the InventoryItem quantity
+* Returns an updated list of InventoryItems
+  * if the warehouseId is not found, will throw 'WarehouseNotFoundException'
+  * if the inventoryItemId is not found, will throw 'inventoryItemNotFoundException'
+
+## 6.11 Get Shipping Report
+* Accepts GET requests to warehouses/:warehouseId/{dateRange}
+* Accepts a warehouseID
+* Returns a json formatted report of transactions
+  * if the warehouseId is not found, will throw 'WarehouseNotFoundException'
```

Comment:
For parameters like a date range, query string parameters would be more standard than path parameters. Path parameters are typically used to identify a specific resource or resources. Query parameters are used for sorting/filtering the requested resources.

For example, a request might look like /warehouses:warehouseId/transactions?date=<whateverdate>